src/main/scala/rocket/RocketCore.scala

```diff
@@ -823,6 +828,25 @@ class Rocket(tile: RocketTile)(implicit p: Parameters) extends CoreModule()(p)
                  wb_reg_wdata))))
   when (rf_wen) { rf.write(rf_waddr, rf_wdata) }
 
+  val ingress_gen = Module(new TraceCoreIngressGen(ingress_params))
```

Comment:
`RocketCoreParams` should have a new flag `enableTraceCoreIngress: Boolean`, defaults to false. If set, it builds this block and connects it as below.